lib/linguist/heuristics.rb

```diff
@@ -160,6 +160,20 @@ def call(data)
       end
     end
 
+    disambiguate "M", "Mathematica", "Matlab", "Mercury", "Objective-C" do |data|
+      if /^[ \t]*@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/.match(data)
+        Language["Objective-C"]
+      elsif data.include?(":- module")
```

Comment:
@arfon , @bkeepers ,
technically, the `:- module` statement doesn't have to be anchored to the begging of the line,
but I guess for all practical purposes it is indeed anchored.
Feel free to add the `^`
Can you also provide a break-down for
https://github.com/Mercury-Language/mercury ?